src/peer_digest.cc

```diff
@@ -597,36 +584,25 @@ peerDigestSwapInMask(void *data, char *buf, ssize_t size)
 }
 
 static int
-peerDigestFetchedEnough(DigestFetchState * fetch, char *buf, ssize_t size, const char *step_name)
+peerDigestFetchedEnough(DigestFetchState * fetch, ssize_t size, const char *step_name)
 {
     static const SBuf hostUnknown("<unknown>"); // peer host (if any)
     SBuf host = hostUnknown;
 
-    PeerDigest *pd = nullptr;
+    const auto pd = fetch->pd.get();
```

Comment:
AFAICT, `pd` may be nil here, but we unconditionally dereference this pointer further below (near `if (!pd->cd)`). It feels like this PR removes too many initial checks from peerDigestFetchedEnough() -- we must check that the peer has not gone way (together with the digest object). Or am I missing some guarantee of `fetch->pd` validity in all peerDigestFetchedEnough() callers?